servant/src/Servant/API/Verbs.hs

```diff
@@ -0,0 +1,166 @@
+type PatchAccepted contentTypes a = Verb 'PATCH 202 contentTypes a
+-- | 'PUT' with 202 status code.
+type PutAccepted contentTypes a = Verb 'PUT 202 contentTypes a
+
+
+-- ** 203 Non-Authoritative Information
+--
+-- Indicates that the request has been successfully processed, but the
+-- information may come from a third-party.
+
+-- | 'GET' with 203 status code.
+type GetNonAuthoritative contentTypes a = Verb 'GET 203 contentTypes a
+-- | 'POST' with 203 status code.
+type PostNonAuthoritative contentTypes a = Verb 'POST 203 contentTypes a
+-- | 'DELETE' with 203 status code.
+type DeleteNonAuthoritative contentTypes a = Verb 'DELETE 203 contentTypes a
+-- | 'PATCH' with 203 status code.
+type PatchNonAuthoritative contentTypes a = Verb 'PATCH 203 contentTypes a
+-- | 'PUT' with 203 status code.
+type PutNonAuthoritative contentTypes a = Verb 'PUT 203 contentTypes a
+
+
+-- ** 204 No Content
+--
+-- Indicates that no response body is being returned. Handlers for these should
+-- return 'NoContent', possibly with headers.
+--
+-- If the document view should be reset, use @205 Reset Content@.
+
+-- | 'GET' with 204 status code.
+type GetNoContent contentTypes noContent = Verb 'GET 204 contentTypes noContent
+-- | 'POST' with 204 status code.
+type PostNoContent contentTypes noContent = Verb 'POST 204 contentTypes noContent
+-- | 'DELETE' with 204 status code.
+type DeleteNoContent contentTypes noContent = Verb 'DELETE 204 contentTypes noContent
+-- | 'PATCH' with 204 status code.
+type PatchNoContent contentTypes noContent = Verb 'PATCH 204 contentTypes noContent
+-- | 'PUT' with 204 status code.
+type PutNoContent contentTypes noContent = Verb 'PUT 204 contentTypes noContent
+
+
+-- ** 205 Reset Content
+--
+-- Indicates that no response body is being returned. Handlers for these should
+-- return 'NoContent', possibly with Headers.
+--
+-- If the document view should not be reset, use @204 No Content@.
+
+-- | 'GET' with 205 status code.
+type GetResetContent contentTypes noContent = Verb 'GET 205 contentTypes noContent
+-- | 'POST' with 205 status code.
+type PostResetContent contentTypes noContent = Verb 'POST 205 contentTypes noContent
+-- | 'DELETE' with 205 status code.
+type DeleteResetContent contentTypes noContent = Verb 'DELETE 205 contentTypes noContent
+-- | 'PATCH' with 205 status code.
+type PatchResetContent contentTypes noContent = Verb 'PATCH 205 contentTypes noContent
+-- | 'PUT' with 205 status code.
+type PutResetContent contentTypes noContent = Verb 'PUT 205 contentTypes noContent
+
+
+-- ** 206 Partial Content
+--
+-- Indicates that the server is delivering part of the resource due to a range
+-- header in the request.
+--
+-- For more information, see <https://tools.ietf.org/html/rfc7233#section-4.1
+-- RFC7233 Section 4.1>
+
+-- | 'GET' with 206 status code.
+type GetPartialContent contentTypes noContent = Verb 'GET 205 contentTypes noContent
+
+
+class ReflectMethod a where
+    reflectMethod :: proxy a -> Method
```

Comment:
I'm not very passionate, except about using the same everywhere in `servant`. I feel `Proxy` is easier to understand. And `proxy` is only more convienient _sometimes_.

I vote for `Proxy` now but I won't stop anyone changing this everywhere in `servant` later.